pkg/querier/queryrange/querysharding.go

```diff
@@ -41,6 +41,7 @@ func NewQueryShardMiddleware(
 	limits Limits,
 	maxShards int,
 	statsHandler queryrangebase.Handler,
+	shardQuantileOverTime bool,
```

Comment:
Same comment as before: this feels like it could be designed better. It feels too specific to special-case one operation like this.

Reply:
I would make the config via configfile only (unless we can easily support multi use or comma separated flag values on cli, I think so but I might be misremembering) and then have a `probabilisticShardingOptions` struct type that we can use to turn each type on or off.